### DISCUSSION TITLE
BPA Generate Script Greyed Out?

### DISCUSSION THREAD

**Discussion by Ahad-boop** [TabularEditor/TabularEditor3#754] [🙋♀️TE3 Q&A] [answered]
Hi! I wanted to understand how to enable the ability to generate script for a fix recommended by BPA. I have a big data model and i have made a copy of it to experiment with these best practises. But i cannot get the generate fix script to work, its greyed out. is there any setting that needs to be changed to allow this to happen?

**Comment by otykier** [comment, marked as answer]
Hi. The fix script needs to be defined in the BpaRules.json file. More info here: https://docs.tabulareditor.com/te2/Best-Practice-Analyzer.html#fixing-objects
